src/com/webonise/polygonassignment/Polygon.java

```diff
@@ -0,0 +1,17 @@
+package com.webonise.polygonassignment;
+
+public abstract class Polygon {
+    protected int noOfSides;
+    protected double lengthOfSides;
+
+    Polygon(int noOfSides) {
+        this.noOfSides = noOfSides;
+    }
+
+    Polygon(int noOfSides, double lengthOfSides) {
+        this.noOfSides = noOfSides;
+        this.lengthOfSides = lengthOfSides;
+    }
+
+    abstract void CalculateResultAndPrint();
+}
```

Comment:
No need to have constructors and member variables inside this class. Just `calculateResultAndPrint()` can be enough.
Need to change class name too. There is no relation between SumOfInternalAnglesCalculator and Polygon.
You can composite these objects inside Polygon class but should not extend.